### PR TITLE
HAAR-686 Limit create user access to authorised roles and add tests

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -34,7 +34,8 @@
         "plugin:prettier/recommended"
       ],
       "rules": {
-        "@typescript-eslint/no-shadow": ["error", { "ignoreTypeValueShadow": true }],
+        "no-shadow": "off",
+        "@typescript-eslint/no-shadow": "error",
```
Comment (karenmillermoj) [inline .eslintrc:38]: This was needed to allow the use of the AuthRole enum

File: backend/middleware/authorisationMiddleware.ts [hunk 2]
```diff
@@ -2,20 +2,37 @@ import jwtDecode from 'jwt-decode'
...
+enum AuthRole {
+  CREATE_USER = 'ROLE_CREATE_USER',
+  MAINTAIN_ACCESS_ROLES_ADMIN = 'ROLE_MAINTAIN_ACCESS_ROLES_ADMIN',
+}
+
+const authorisationMap = {
+  '/select-caseloads': AuthRole.MAINTAIN_ACCESS_ROLES_ADMIN,
+  '/create-dps-user': AuthRole.CREATE_USER,
+  '/create-user': AuthRole.CREATE_USER,
+}
```
Comment (karenmillermoj) [inline backend/middleware/authorisationMiddleware.ts:14]: All pages and their appropriate role can just be added into this map.

**Review by brightonsbox — APPROVED** [review]
LGTM

File: integration-tests/integration/dpsuser.create.spec.js [hunk 3]
```diff
@@ -110,8 +111,29 @@ context('DPS user create functionality', () => {
...
+    const menuPage = MenuPage.verifyOnPage()
+
+    menuPage.createDpsUserTile().should('not.exist')
+
+    cy.visit('/create-dps-user', { failOnStatusCode: false })
+    AuthErrorPage.verifyOnPage()
+  })
   it('Should check for CSRF token', () => {
-    cy.task('stubSignIn', { roles: [{ roleCode: 'CREATE_USER' }] })
+    cy.task('stubSignIn', { roles: [{ roleCode: 'CREATE_USER' }], tokenRoles: ['ROLE_CREATE_USER'] })
```
Comment (karenmillermoj) [inline integration-tests/integration/dpsuser.create.spec.js:136]: roles and tokenRoles could possibly be combined, it's just that the tokenRoles is also expecting GLOBAL_SEARCH - I didn't want to change this functionality for this PR

**Pull request merged**